### PR TITLE
Laravel 9 support

### PR DESCRIPTION
**Pull request by phpsa**
* [x] removed deprecated listener as is now core of laravel
* [x] fixed db macros orWhereNotNullOrEmpty
* [x] Added orFail Relationship macro
* [x] deprecated execute helper (app(XXX) is better)

### CODE REVIEW

File: tests/DatabaseMacrosTest.php [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        $this->loadMigrationsFrom(__DIR__ . '/migrations');
+    }
+
+    // protected function defineDatabaseMigrations()
+    // {
+
+    //     $this->loadMigrationsFrom(__DIR__.'/../migrations');
+
+    //     $this->artisan('migrate', ['--database' => 'testbench'])->run();
+    // }
```
Comment (robertatcd) [inline tests/DatabaseMacrosTest.php]: Commented-out code.

**Review by robertatcd — COMMENTED** [review]
Looks good to me @phpsa . How would I go about testing it in my project?

**Comment by phpsa** [comment]
> Looks good to me @phpsa . How would I go about testing it in my project?

@robertatcd - pushed release 3 which you can plugin to your project.
make sure to read the upgrade notes

**Comment by samatcd** [comment]
This looks really good @phpsa. Might make a good contribution to Laravel! I know I'd like to have this functionality in several projects I'm working on.

**Pull request merged**